tests/framework/db/mysql/SchemaTest.php

```diff
@@ -38,7 +38,33 @@ public function testLoadDefaultDatetimeColumn()
 
         $dt = $schema->columns['dt'];
 
-        $this->assertInstanceOf(Expression::className(),$dt->defaultValue);
+        $this->assertInstanceOf(Expression::className(), $dt->defaultValue);
+        $this->assertEquals('CURRENT_TIMESTAMP', (string)$dt->defaultValue);
+    }
+
+    public function testDefaultDatetimeColumnWithMicrosecs()
+    {
+        if (!version_compare($this->getConnection()->pdo->getAttribute(\PDO::ATTR_SERVER_VERSION), '5.6.4', '>=')) {
+            $this->markTestSkipped('CURRENT_TIMESTAMP with microseconds as default column value is supported since MySQL 5.6.4.');
+        }
+        $sql = <<<SQL
+CREATE TABLE  IF NOT EXISTS `current_timestamp_test`  (
+  `dt` datetime(2) NOT NULL DEFAULT CURRENT_TIMESTAMP(2),
+  `ts` timestamp(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8
+SQL;
+
+        $this->getConnection()->createCommand($sql)->execute();
+
+        $schema = $this->getConnection()->getTableSchema('current_timestamp_test');
+
+        $dt = $schema->columns['dt'];
+        $this->assertInstanceOf(Expression::className(), $dt->defaultValue);
+        $this->assertEquals('CURRENT_TIMESTAMP(2)', (string)$dt->defaultValue);
+
+        $ts = $schema->columns['ts'];
+        $this->assertInstanceOf(Expression::className(), $ts->defaultValue);
+        $this->assertEquals('CURRENT_TIMESTAMP(3)', (string)$ts->defaultValue);
```

Comment:
Is there a reason to use two nearly equal cases?

Reply:
It's in case loading process for different type of column returns the expected value. Number here doesn't really matter. And current_timestamp without microseconds is tested in previous method.

Reply:
So lets add one more column with `ON UPDATE` part and resolve this :+1:

Reply:
There is ON UPDATE tested in method above :)

Reply:
I meant to check working with microseconds AND `on update`, not just one of them :)